22304-vmtarget-pharo-not-found-during-bootstrap

akgrant43 opened the pull request:
vmtarget/pharo is referenced inconsistently resulting in the executable
not being found depending on the values of BOOTSTRAP_REPOSITORY and
BOOTSTRAP_CACHE.

- Move vmtarget to ${CACHE}
  This also simplifies tidying up afterwards as removing ${CACHE} will
  also remove vmtarget.
- Modify all references to vmtarget/pharo to use ${CACHE}

Fogbugz: https://pharo.fogbugz.com/f/cases/22304/vmtarget-pharo-not-found-during-bootstrap

akgrant committed:
22304-vmtarget-pharo-not-found-during-bootstrap

vmtarget/pharo is referenced inconsistently resulting in the executable
not being found depending on the values of BOOTSTRAP_REPOSITORY and
BOOTSTRAP_CACHE.

- Move vmtarget to ${CACHE}
  This also simplifies tidying up afterwards as removing ${CACHE} will
  also remove vmtarget.
- Modify all references to vmtarget/pharo to use ${CACHE}

akgrant43 commented:
CI test failures look unrelated - they're network errors after the build completes.

akgrant committed:
22304: Update bootstrap/scripts/snapshot.sh

to correctly reference the bootstrap cache directory.

akgrant43 commented:
CI failures are again network related.

The latest commit (b20611f) isn't used during the build process, so couldn't have any affect.

akgrant committed:
22304: consolidate env vars CACHE and BOOTSTRAP_CACHE in build scripts

Remove the environment variable CACHE and just use BOOTSTRAP_CACHE.
Previously CACHE would be set to either BOOTSTRAP_CACHE or, if absent, a
default value.  Now BOOTSTRAP_CACHE is set to the default value if it is
empty.

akgrant43 commented:
CI failures are again network related.

pavel-krivanek commented on bootstrap/scripts/snapshot.sh:
```diff
@@ -12,5 +12,5 @@ set -e
 
 pushd "${BOOTSTRAP_REPOSITORY}"
-tar czf "${BOOTSTRAP_REPOSITORY}/snapshot.tar.gz" Pharo.image Pharo.changes pharo pharo-ui pharo-vm pharo-local vmtarget bootstrap-cache
+tar czf "${BOOTSTRAP_REPOSITORY}/snapshot.tar.gz" Pharo.image Pharo.changes pharo pharo-ui pharo-vm pharo-local ${CACHE}
```
it should be BOOTSTRAP_CACHE here

akgrant43 replied:
Oops, sorry about that.

Actually snapshot.sh and restore.sh were both assuming that bootstrap.sh was being run from within the respository.  Now they should be run under the same conditions as bootstrap.sh.

akgrant committed:
22304: Update snapshot.sh and restore.sh

Both scripts originally assumed they were being run from the repository
root directory.  Both scripts must now be run under the same conditions
as bootstrap.sh (cwd, BOOTSTRAP_REPOSITORY, BOOTSTRAP_CACHE).

akgrant committed:
22304 bootstrap scripts still not completely directory independent

The scripts still made assumptions about where they were executed from
when trying to run envvars.sh.

Assume that all scripts are in the same directory.

akgrant43 commented:
CI failures unrelated: same 3 ZnHTTPS tests

akgrant43 commented:
Wow, that was fast!  Thanks Pavel & Marcus!
